src/MaxMind/Db/Reader/Util.php

```diff
@@ -11,7 +11,7 @@ class Util
      */
     public static function read($stream, int $offset, int $numberOfBytes): string
     {
-        if ($numberOfBytes === 0) {
+        if ($numberOfBytes <= 0) {
```

Comment:
Maybe instead we could add a type hint for `$numberOfBytes` in the params. I think it would be a bug if a negative number was passed to this function.

Reply:
If a negative value is a bug, maybe it would be better to throw an exception?

```php
        if ($numberOfBytes === 0) {
            return '';
        } elseif ($numberOfBytes < 0) {
            throw new InvalidDatabaseException(
                'The number of bytes may not be a negative value'
            );
        }
```
Is this what you are looking for?
```php
    /**
     * @param resource $stream
     * @param int $offset
     * @param int<0, max> $numberOfBytes
     * @return string
     */
```

A little warning, that once we do this change, then we end up with a follow-up error from the caller Decoder.php:
```
 ------ ------------------------------------------------------------------------------------------------------------- 
  Line   src/MaxMind/Db/Reader/Decoder.php                                                      
 ------ ------------------------------------------------------------------------------------------------------------- 
  125    Parameter #3 $numberOfBytes of static method MaxMind\Db\Reader\Util::read() expects int<0, max>, int given.  
```

Which I need to address next.

Reply:
Where we can have PHPStan check for us, I think that would be preferable to the run-time cost of the check. However, I am not opposed to adding the check for now.

Reply:
Done.